<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Kajetan Åšwerik</author>
 <class>mainWindowClass</class>
 <widget class="QMainWindow" name="mainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>474</width>
    <height>487</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Morph OpenCL</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="4">
       <widget class="QGroupBox" name="gbElement">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Structuring element</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QRadioButton" name="rbRect">
           <property name="text">
            <string>Rectangle</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbEllipse">
           <property name="text">
            <string>Ellipse</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbCross">
           <property name="text">
            <string>Cross</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbDiamond">
           <property name="text">
            <string>Diamond</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pbShowSE">
           <property name="text">
            <string>Show structuring element</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QGroupBox" name="gbElementSize">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Element size</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0" colspan="2">
          <widget class="QCheckBox" name="cbSquare">
           <property name="text">
            <string>1:1 ratio</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QSlider" name="hsXElementSize">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>75</number>
           </property>
           <property name="pageStep">
            <number>2</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksBelow</enum>
           </property>
           <property name="tickInterval">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="lbXElementSize">
           <property name="text">
            <string/>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QSlider" name="hsYElementSize">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>75</number>
           </property>
           <property name="pageStep">
            <number>2</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksBelow</enum>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="lbYElementSize">
           <property name="text">
            <string/>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="5" column="0" rowspan="3">
          <widget class="QDial" name="dialRotation">
           <property name="minimumSize">
            <size>
             <width>64</width>
             <height>64</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>64</width>
             <height>64</height>
            </size>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="value">
            <number>180</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="wrapping">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLabel" name="lbRotation">
           <property name="text">
            <string/>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QPushButton" name="pbResetRotation">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="3" rowspan="2">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="bpOperation">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Operation</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <property name="spacing">
            <number>6</number>
           </property>
           <item row="0" column="0" colspan="2">
            <widget class="QRadioButton" name="rbNone">
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QCheckBox" name="cbInvert">
             <property name="text">
              <string>Invert</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="Line" name="line">
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="lineWidth">
              <number>1</number>
             </property>
             <property name="midLineWidth">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="10" column="0" colspan="2">
            <widget class="QRadioButton" name="rbErode">
             <property name="text">
              <string>Erode</string>
             </property>
            </widget>
           </item>
           <item row="11" column="0" colspan="2">
            <widget class="QRadioButton" name="rbDilate">
             <property name="text">
              <string>Dilate</string>
             </property>
            </widget>
           </item>
           <item row="12" column="0" colspan="2">
            <widget class="QRadioButton" name="rbOpen">
             <property name="text">
              <string>Open</string>
             </property>
            </widget>
           </item>
           <item row="13" column="0" colspan="2">
            <widget class="QRadioButton" name="rbClose">
             <property name="text">
              <string>Close</string>
             </property>
            </widget>
           </item>
           <item row="15" column="0" colspan="2">
            <widget class="QRadioButton" name="rbTopHat">
             <property name="text">
              <string>Top Hat</string>
             </property>
            </widget>
           </item>
           <item row="17" column="0" colspan="2">
            <widget class="QRadioButton" name="rbBlackHat">
             <property name="text">
              <string>Black Hat</string>
             </property>
            </widget>
           </item>
           <item row="20" column="0" colspan="2">
            <widget class="QRadioButton" name="rbSkeleton">
             <property name="text">
              <string>Skeletonize (8-connectivity)</string>
             </property>
            </widget>
           </item>
           <item row="19" column="0" colspan="2">
            <widget class="QRadioButton" name="rbOutline">
             <property name="text">
              <string>Outline</string>
             </property>
            </widget>
           </item>
           <item row="18" column="0">
            <widget class="QRadioButton" name="rbGradient">
             <property name="text">
              <string>Morphological gradient</string>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="QPushButton" name="pbRun">
             <property name="text">
              <string>Run</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0" colspan="2">
            <widget class="QCheckBox" name="cbAutoTrigger">
             <property name="text">
              <string>Run on every change</string>
             </property>
            </widget>
           </item>
           <item row="9" column="0" colspan="2">
            <widget class="Line" name="line_2">
             <property name="lineWidth">
              <number>1</number>
             </property>
             <property name="midLineWidth">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="21" column="0">
            <widget class="QRadioButton" name="rbSkeletonZhang">
             <property name="text">
              <string>Skeletonize (Zhang and Suen)</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QComboBox" name="cmbBayer">
             <item>
              <property name="text">
               <string>No preprocessing</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bayer RG</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bayer GR</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bayer BG</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bayer GB</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="3" colspan="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0" rowspan="3">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="GLWidget" name="glWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>10</width>
            <height>10</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::MinimumExpanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::MinimumExpanding</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>474</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionCameraInput"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionSaveSE"/>
    <addaction name="actionOpenSE"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionOpenCL"/>
    <addaction name="actionPickMethod"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSettings"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open image</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save image</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionUse_OpenCL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use OpenCL</string>
   </property>
  </action>
  <action name="actionOpenCL_Settings">
   <property name="text">
    <string>OpenCL Settings</string>
   </property>
  </action>
  <action name="actionOpenCL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use OpenCL</string>
   </property>
   <property name="toolTip">
    <string>Use OpenCL instead of OpenCV</string>
   </property>
  </action>
  <action name="actionPickMethod">
   <property name="text">
    <string>Pick method</string>
   </property>
   <property name="toolTip">
    <string>Pick method again (buffer 1D or 2D)</string>
   </property>
  </action>
  <action name="actionCameraInput">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>From camera</string>
   </property>
  </action>
  <action name="actionSaveSE">
   <property name="text">
    <string>Save structuring element</string>
   </property>
  </action>
  <action name="actionOpenSE">
   <property name="text">
    <string>Open structuring element</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>rbNone</tabstop>
  <tabstop>cbInvert</tabstop>
  <tabstop>rbErode</tabstop>
  <tabstop>rbDilate</tabstop>
  <tabstop>rbOpen</tabstop>
  <tabstop>rbClose</tabstop>
  <tabstop>rbTopHat</tabstop>
  <tabstop>rbBlackHat</tabstop>
  <tabstop>rbOutline</tabstop>
  <tabstop>rbSkeleton</tabstop>
  <tabstop>rbRect</tabstop>
  <tabstop>rbEllipse</tabstop>
  <tabstop>rbCross</tabstop>
  <tabstop>rbDiamond</tabstop>
  <tabstop>cbSquare</tabstop>
  <tabstop>hsXElementSize</tabstop>
  <tabstop>hsYElementSize</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
